plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftSparkMetastoreUtil.java

```diff
@@ -0,0 +1,260 @@
+                "spark.sql.statistics.colStats.c_bool.maxLen", "1",
+                "spark.sql.statistics.colStats.c_bool.avgLen", "1",
+                "spark.sql.statistics.colStats.c_bool.nullCount", "1",
+                "spark.sql.statistics.colStats.c_bool.version", "2",
+                "spark.sql.statistics.colStats.c_bool.max", "true");
+        HiveColumnStatistics actualNotExists = fromMetastoreColumnStatistics(new FieldSchema("c_bool_not_exists", BOOLEAN_TYPE_NAME, null), columnStatistics, OptionalLong.of(10));
+        assertEquals(actualNotExists.getIntegerStatistics(), Optional.empty());
+        assertEquals(actualNotExists.getDoubleStatistics(), Optional.empty());
+        assertEquals(actualNotExists.getDecimalStatistics(), Optional.empty());
+        assertEquals(actualNotExists.getDateStatistics(), Optional.empty());
+        assertEquals(actualNotExists.getBooleanStatistics(), Optional.of(new BooleanStatistics(OptionalLong.empty(), OptionalLong.empty())));
+        assertEquals(actualNotExists.getMaxValueSizeInBytes(), OptionalLong.empty());
+        assertEquals(actualNotExists.getTotalSizeInBytes(), OptionalLong.empty());
+
+        HiveColumnStatistics actual = fromMetastoreColumnStatistics(new FieldSchema("c_bool", BOOLEAN_TYPE_NAME, null), columnStatistics, OptionalLong.of(10));
+        assertEquals(actual.getIntegerStatistics(), Optional.empty());
+        assertEquals(actual.getDoubleStatistics(), Optional.empty());
+        assertEquals(actual.getDecimalStatistics(), Optional.empty());
+        assertEquals(actual.getDateStatistics(), Optional.empty());
+        assertEquals(actual.getBooleanStatistics(), Optional.of(new BooleanStatistics(OptionalLong.empty(), OptionalLong.empty())));
+        assertEquals(actual.getMaxValueSizeInBytes(), OptionalLong.empty());
+        assertEquals(actual.getTotalSizeInBytes(), OptionalLong.empty());
+        assertEquals(actual.getNullsCount(), OptionalLong.of(1));
+        assertEquals(actual.getDistinctValuesCount(), OptionalLong.empty());
+    }
+
+    @Test
+    public void testSparkDateStatsToColumnStatistics()
+    {
+        Map<String, String> columnStatistics = Map.of("spark.sql.statistics.colStats.c_date.min", "2000-01-01",
+                "spark.sql.statistics.colStats.c_date.distinctCount", "10",
+                "spark.sql.statistics.colStats.c_date.maxLen", "4",
+                "spark.sql.statistics.colStats.c_date.avgLen", "4",
+                "spark.sql.statistics.colStats.c_date.nullCount", "3",
+                "spark.sql.statistics.colStats.c_date.version", "2",
+                "spark.sql.statistics.colStats.c_date.max", "2030-12-31");
+        HiveColumnStatistics actualNotExists = fromMetastoreColumnStatistics(new FieldSchema("c_date_not_exists", DATE_TYPE_NAME, null), columnStatistics, OptionalLong.of(10));
+        assertEquals(actualNotExists.getIntegerStatistics(), Optional.empty());
```

Comment:
There is a lot of repetition here.
Group the assertions in one method to reuse them.

You could have a `Builder` class for `HiveColumnStatistics` instances where you could fill in only one specific stat (e.g.  `builder().withDateStatistics(new DateStatistics(Optional.empty(), Optional.empty()))` in your example below).

This way `9` lines of assertion code get reduced to `1`/`2` lines of code.